src/scipp/html/formatting_datagroup_html.py

```diff
@@ -11,25 +12,58 @@
 from ..core.cpp_classes import DataArray, Dataset, Variable
 from ..core.data_group import DataGroup
 from ..units import dimensionless
-from .resources import load_atomic_tpl, load_collapsible_tpl, load_dg_detail_tpl, \
-    load_dg_repr_tpl, load_dg_style
+from .resources import load_atomic_row_tpl, load_collapsible_row_tpl, \
+    load_dg_detail_list_tpl, load_dg_repr_tpl, load_dg_style
 
 
-def _make_shape_repr(var: Union[Variable, DataArray, Dataset, DataGroup]) -> str:
+def _format_shape(var: Union[Variable, DataArray, Dataset, DataGroup]) -> str:
     """Returns HTML Component that represents the shape of ``var``"""
     shape_list = "".join(f"<li>{escape(str(dim))}: {size}</li>"
                          for dim, size in var.sizes.items())
     return f"<ul class='sc-dim-list'>{shape_list}</ul>"
 
 
-def _make_dataset_summary(ds: Dataset) -> str:
-    """Returns (partial) information of Dataset object items"""
-    key_type_str = [
-        ": ".join([name, type(value).__name__]) for name, value in ds.items()
-    ]
-    if len(key_type_str) > 3:
-        key_type_str = key_type_str[:2] + ["..."] + key_type_str[-1:]
-    return "Dataset{" + ", ".join(key_type_str) + "}"
+def _format_atomic_value(value, maxlen: int = 5) -> str:
+    value_repr = str(value)[:maxlen]
+    if len(value_repr) < len(str(value)):
+        value_repr += "..."
+    return value_repr
+
+
+def _format_dictionary_item(name_item: tuple, maxlen: int = 10) -> str:
+    name, item = name_item
+    name = _format_atomic_value(name, maxlen=maxlen)
+    type_repr = _format_atomic_value(type(item).__name__, maxlen=maxlen)
+    return "(" + ": ".join((name, type_repr)) + ")"
+
+
+def _format_multi_dim_data(var: Union[Variable, DataArray, Dataset, np.ndarray],
+                           max_item_number: int = 2) -> str:
+    if len(var.shape) == 0 and (not isinstance(var, Dataset)):
+        return _format_atomic_value(var.value, maxlen=30)
+    elif isinstance(var, Dataset):
+        view_iter = iter(var.items())
+        format_item = _format_dictionary_item
+        var_len = len(var)
+    elif isinstance(var, (Variable, DataArray)):
+        view_iter = iter(np.ravel(var.values))
+        format_item = _format_atomic_value
+        var_len = len(var)
+    else:
+        view_iter = iter(np.ravel(var))
+        format_item = _format_atomic_value
+        var_len = var.size
+
+    max_item_number = max(2, max_item_number)
+    max_first_items = min(var_len, max_item_number - 1)
+    view_items = [format_item(next(view_iter)) for _ in range(max_first_items)]
+
+    if var_len > max_first_items:
+        if var_len > max_item_number:
+            view_items.append('... ')
+        view_items.append(format_item(deque(view_iter).pop()))
+
+    return ', '.join(view_items)
```

Comment:
The changes in this file mean we get more inline previews a values now?

Reply:
Yes, not it also shows preview of `DataArray` and `Variable` now